### DISCUSSION TITLE
How to use it in Power Apps?

### DISCUSSION THREAD

**Discussion by xunyicjq** [microsoft/semantic-kernel#1087] [General]
Is there any Sample for Power Apps to use ?

**Comment by dluc** [comment]
@evchaki FYI

**Comment by craigomatic** [comment]
> Is there any Sample for Power Apps to use ?

Would the use case here be a Canvas app that calls out to a function that hosts SK? Something else?

**Comment by xunyicjq** [comment]
Yes.
How to call SK in Canvas app.
Is there any Sample ?

**Comment by xunyicjq** [comment]
Yes.
How to call SK in Canvas app.
Is there any Sample ?

**Comment by evchaki** [comment]
@xunyicjq ,  here are the steps, but there is some work for you to do to get it all setup:

1. Deploy Semantic Kernel to Azure
2. Once in Azure you need to create a Swagger / OpenAPI spec for the Kernel and for the skills you want to use in Power Apps / Power Automate (You can use API Management for this)
3.  On the go to the API definition tab - if you deployed Semantic Kernel as a web app and put in the URL to the JSON for the Swagger definition.  There is a button on that page that will create you a custom connector.
4. Go to Power Apps or Power Automate and select the environment you pushed the custom connector to
5. Create a new app or flow and use that custom connector.  You should be able to pass the ask over and get results.

**Reply by SeaDude** [reply]
When you say "Deploy Semantic Kernel to Azure", how exactly would you do this? What resource type would host SK?

Thank you

**Reply by evchaki** [reply]
You can do it this way - https://devblogs.microsoft.com/semantic-kernel/how-to-deploy-semantic-kernel-to-azure-as-a-web-application-service/ or this way https://devblogs.microsoft.com/semantic-kernel/how-to-deploy-semantic-kernel-to-azure-in-minutes/ - depending on your needs.

**Reply by SeaDude** [reply]
Ah, ok. I like the option of hosting SK as an API using an Azure Function. (Your second option).

Thanks for the write up.

**Comment by pschandekar09** [comment]
@evchaki How can I integrate power app dataverse endpoint as skill into semantic kernel?

**Reply by evchaki** [reply]
you will need to call the end point to your Dataverse environment and grab the API (https://learn.microsoft.com/en-us/power-apps/developer/data-platform/webapi/query-data-web-api)  from that.  You can try to use the OpenAPI skill (https://github.com/microsoft/semantic-kernel/tree/main/dotnet/src/Skills/Skills.OpenAPI) we have.

**Reply by pschandekar09** [reply]
Actually, I would like to implement a semantic copilot chat. It should be able to look into the data of Dataverse and provide me with the answer. Additionally, I would like to integrate other skills, such as MS-Graph and other unstructured data from documents. Could you please guide me on how to implement this? Do you have any blogs or references that I can refer to?

**Reply by evchaki** [reply]
Perfect, start with our copilot chat starter app (https://github.com/microsoft/semantic-kernel/tree/main/samples/apps/copilot-chat-app & https://learn.microsoft.com/en-us/semantic-kernel/samples-and-solutions/copilot-chat).  You can see how we are calling out to other connectors in the app.

**Reply by pschandekar09** [reply]
Yes, I have already run that sample application on my local machine. Now, I want to explore more on integrating Power Apps, Dataverse, MS Graph, etc. So please guide me on this how to integrate it.
